### PR TITLE
WT-8428 Add a test regarding the release_evict_page option

### PR DESCRIPTION
**Pull request by etienneptl**
In [WT-8208](https://jira.mongodb.org/browse/WT-8208), we added a new session debug configuration `release_evict_page` which evicts pages as they are released. This PR adds a test dedicated to testing this new feature.
The test was first added as part of #7172, but as has been slightly modified in this PR.

### CODE REVIEW

File: test/suite/test_config11.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    scenarios = make_scenarios(key_format_values)
+
+    def test_config11(self):
+        uri = 'table:test_config11'
+
+        ds = SimpleDataSet(
+            self, uri, 0, key_format=self.key_format, value_format="S", config='log=(enabled=false)')
+        ds.populate()
+
+        # Pin oldest and stable timestamps to 10.
```
Comment (etienneptl) [inline test/suite/test_config11.py:56]: @sulabhM @mm-ng, I am not sure why is this necessary?

File: test/suite/test_config11.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        # Insert some values to fill up the cache, but not exceeding 80% of it (we set a margin of
+        # 10% to be sure).
+        value = 'abcd' * 1000
+        current_cache_usage_perc = 0
+        threshold_cache_perc = 70
+        i = 1
+
+        s = self.conn.open_session()
+        cursor = s.open_cursor(uri)
+        while(current_cache_usage_perc < threshold_cache_perc):
```
Comment (etienneptl) [inline test/suite/test_config11.py:74]: I have modified this condition so we get to fill the cache until a certain point. Before we were using `nrows` but it wouldn't give us a real indication about the cache size being used.

File: test/suite/test_config11.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+            # Retrieve the current cache usage.
+            current_cache_usage_perc = stat_cursor[stat.conn.cache_bytes_inuse][2] * 100 / max_cache_size
+            stat_cursor.reset()
+
+        cursor.close()
+
+        # Checkpoint should not trigger eviction, it still might reduce the cache usage by a small
+        # amount.
+        s.checkpoint()
```
Comment (etienneptl) [inline test/suite/test_config11.py:86]: @sulabhM @mm-ng , I saw in the first version of the test, we perform a checkpoint. Was there any specific reason for doing a checkpoint?
Since we don't expect eviction to be triggered, I added a test just below. We might as well remove it if performing checkpoint is not required for this test.

File: test/suite/test_config11.py (hunk 1 above)
Comment (sulabhM) [reply test/suite/test_config11.py:56]: I dont remember well, maybe we were trying to make the test timestamped. In that case we would have inserted multiple values for the same key. To make sure older values are sent to history store on eviction and not thrown away, we would pin the oldest.

File: test/suite/test_config11.py (hunk 3 above)
Comment (sulabhM) [reply test/suite/test_config11.py:86]: To make the content is the cache clean. If you don't checkpoint, the cache is all dirty and eviction will trigger at 5% dirty by itself. We wanted to fill the cache with clean content, such that we can use the newly added flag to trigger eviction.

Comment (etienneptl) [reply test/suite/test_config11.py:86]: So when we are filling the cache close to 80%, it's already too late isn't it? The threshold of 5% is exceeded very quickly and eviction for dirty content is already being executed.
Should I fill the cache up to 5% to avoid this at the beginning of the test?

File: test/suite/test_config11.py (hunk 1 above)
Comment (etienneptl) [reply test/suite/test_config11.py]: As discussed, I removed any ts related operation as part of 4674012995baa58335e740454fa99f1cc1062a1e

File: test/suite/test_config11.py (hunk 3 above)
Comment (etienneptl) [reply test/suite/test_config11.py:78]: I have explicitly configured a higher threshold for dirty content [here](https://github.com/wiredtiger/wiredtiger/pull/7515/commits/2052a9bf80544fe37874b38a576f054292930841#diff-f1e790183032bc0e6d3042a62d62525a2867f3f486849e65da212b1e5e0443d6R40).

**Review by sulabhM — APPROVED** [review]
Please run the test a loop for a few hours to be sure that it always passes.

**Review by mm-ng — APPROVED** [review]
LGTM
